pcweb/pages/docs/recipes/filtered_table.py

```diff
@@ -0,0 +1,117 @@
+import reflex as rx
+
+from pcweb.templates.docpage import doccode, docheader, docpage, doctext, subheader
+
+from pcweb.base_state import State
+
+
+code_example = """
+import reflex as rx
+from typing import List, Dict
+
+RAW_DATA = [
+    {"name": "Alice", "tags": "tag1"},
+    {"name": "Bob", "tags": "tag2"},
+    {"name": "Charlie", "tags": "tag1"},
+]
+RAW_DATA_COLUMNS = ["Name", "tags"]
+
+
+class FilteredTableState(rx.State):
+    filter_expr: str = ""
+    data: Dict[str, Dict[str, str]] = RAW_DATA
+
+    @rx.cached_var
+    def filtered_data(self) -> List[Dict[str, str]]:
+        # Use this generated filtered data view in the `rx.foreach` of
+        #  the table renderer of rows
+        # It is dependent on `filter_expr`
+        # This `filter_expr` is set by an rx.input
+        return [
+            row
+            for row in self.data
+            if self.filter_expr == ""
+            or self.filter_expr != ""
+            and self.filter_expr == row["tags"]
+        ]
+
+    def input_filter_on_change(self, value):
+        self.filter_expr = value
+        # for DEBUGGING
+        yield rx.console_log(f"Filter set to: {self.filter_expr}")
+
+
+def render_row(row):
+    return rx.tr(rx.td(row["name"]), rx.td(row["tags"]))
+
+
+def render_rows():
+    return [
+        rx.foreach(
+            # use data filtered by `filter_expr` as update by rx.input
+            FilteredTableState.filtered_data,
+            render_row,
+        )
+    ]
+
+
+def render_table():
+    return rx.table_container(
+        rx.table(
+            rx.thead(rx.tr(*[rx.th(column) for column in RAW_DATA_COLUMNS])),
+            rx.tbody(*render_rows()),
+        )
+    )
+
+
+def index() -> rx.Component:
+    return rx.box(
+        rx.box(
+            rx.heading(
+                "Filter by tags:",
+                size="sm",
+            ),
+            rx.debounce_input(
```

Comment:
```suggestion
```

If you only want to set the `debounce_timeout`, that is actually a prop on `rx.input` now as well. Under the hood it will automatically wrap the input in a `debounce_input`.

Reply:
Thank you for these suggestions and pointers, invaluable learning in this way